src/rest.cpp

```diff
@@ -260,15 +277,15 @@ static bool rest_tx(AcceptedConnection* conn,
     vector<string> params;
     const RetFormat rf = ParseDataFormat(params, strReq);
 
-    string hashStr = params[0];
+    CheckRequestString(params[0], IT_TXHASH);
```

Comment:
Isn't this pre-check redundant? e.g., ParseHash will already fail on invalid hashes.